packages/features/ee/payments/components/Payment.tsx

```diff
@@ -53,21 +65,24 @@ const PaymentForm = (props: Props) => {
 
     let payload;
     const params: { [k: string]: any } = {
-      uid: props.bookingUid,
+      uid: props.booking.uid,
       email: router.query.email,
     };
+
+    const returnUrl = getReturnUrl(props);
+
     if (paymentOption === "HOLD" && "setupIntent" in props.payment.data) {
       payload = await stripe.confirmSetup({
         elements,
         confirmParams: {
-          return_url: props.eventType.successRedirectUrl || `${CAL_URL}/booking/${props.bookingUid}`,
+          return_url: returnUrl.toString(),
```

Comment:
And use shorthand notation here.
```suggestion
          return_url,
```